### PR TITLE
Added support for atonement tracking and early atonement application.…

### PR DESCRIPTION
**Pull request by nutspanther**
… Updated the spell id for Atonement as it was incorrect. Added in a Statistic Box and Issue tracker for Atonements on the parser.

### CODE REVIEW

**Comment by MartijnHols** [comment]
Hey thanks for the pull request, that's awesome! :D I'm a bit new to working with PRs so please bear with me.

I made a few small changes mostly to code style in this branch: https://github.com/MartijnHols/WoWAnalyzer/commits/nutspanther-master

There are two noteworthy things to discuss:

1. You updated the Atonement buff id, did you do this knowing that there are actually **two** different spell ids for Atonement which both appear to be active? I haven't figured out yet what causes this, but I assume they're both important to track so I did that in 2098a7dee442b7a1b949bd62d9d9e47f72eeb4c9. Let me know if you think this is wrong.
2. I'm not sure showing the refreshes as percentage is the clearest for users. Seeing "9.09%" has little impact to me, I'd prefer seeing the absolute number 18 as that makes it appear more important. I know this sounds a bit weird since a percentage is more informative and accurate, but I feel like showing the exact number of times you fucked something up is more useful than a percentage that feels like it's decent even if it isn't.

Also it would be great if the tooltip could includes the total amount of Atonement refreshes too (including those within 3 seconds of it expiring).

**Comment by nutspanther** [comment]
1. I did not know that there were 2 different id's for it. I just changed the one because the other listed one wasn't working for me. I may need to look into it in more detail.
2. I have the numbers in the tooltip, can definitely reverse them and add in the count for general refreshes.
    1. Total Atonement Applications
    2. Total Atonement Refreshes
    3. Total Improper Atonement Refreshes

**Comment by MartijnHols** [comment]
That would be nice! Will merge once those changes are in.

**Comment by nutspanther** [comment]
Updated with your requests. I am unable to pull in the changes that you made on nutspanther-master, so I suppose you would need to merge those in to the pull request?

**Comment by MartijnHols** [comment]
That's alright. I'll be looking into it now. For future reference I think you should be able to do this:

```
git checkout -b wowanalyzer-nutspanther-master master
git pull https://github.com/MartijnHols/WoWAnalyzer.git nutspanther-master
```

**Comment by nutspanther** [comment]
Would you look at that...okay will do that in the future. My Git kung-fu could be better.

**Comment by nutspanther** [comment]
Went ahead and merged and pushed the changes. You shouldn't have to now.

File: src/Parser/DisciplinePriest/CombatLogParser.js [hunk 1]
```diff
@@ -170,6 +182,17 @@ class CombatLogParser extends MainCombatLogParser {
           </dfn>
         )}
       />,
+      this.modules.atonement.active && (
+        <StatisticBox 
+        icon={<SpellIcon id={SPELLS.ATONEMENT1.id} />}
+        value= {this.modules.atonement.improperAtonementRefreshes.length}
+        label={(
+          <dfn data-tip={`${((this.modules.atonement.improperAtonementRefreshes.length / this.modules.atonement.totalAtonementRefreshes * 100) || 0).toFixed(2)}% (${this.modules.atonement.improperAtonementRefreshes.length}/${this.modules.atonement.totalAtonementRefreshes}) of your Atonement refreshes were too early. ${((this.modules.atonement.improperAtonementRefreshes.length / this.modules.atonement.totalAtones * 100) || 0).toFixed(2)}% (${this.modules.atonement.improperAtonementRefreshes.length}/${this.modules.atonement.totalAtones}) of all Atonements applied were refreshed too early.` }>
+             Early Atonement Refreshes
```
Comment (MartijnHols) [inline src/Parser/DisciplinePriest/CombatLogParser.js]: Lower case R (nit picky, sorry, but consistency!)

File: src/Parser/DisciplinePriest/CombatLogParser.js [hunk 2]
```diff
@@ -170,6 +182,17 @@ class CombatLogParser extends MainCombatLogParser {
           </dfn>
         )}
       />,
+      this.modules.atonement.active && (
+        <StatisticBox 
+        icon={<SpellIcon id={SPELLS.ATONEMENT1.id} />}
+        value= {this.modules.atonement.improperAtonementRefreshes.length}
+        label={(
+          <dfn data-tip={`${((this.modules.atonement.improperAtonementRefreshes.length / this.modules.atonement.totalAtonementRefreshes * 100) || 0).toFixed(2)}% (${this.modules.atonement.improperAtonementRefreshes.length}/${this.modules.atonement.totalAtonementRefreshes}) of your Atonement refreshes were too early. ${((this.modules.atonement.improperAtonementRefreshes.length / this.modules.atonement.totalAtones * 100) || 0).toFixed(2)}% (${this.modules.atonement.improperAtonementRefreshes.length}/${this.modules.atonement.totalAtones}) of all Atonements applied were refreshed too early.` }>
```
Comment (MartijnHols) [inline src/Parser/DisciplinePriest/CombatLogParser.js]: Is the percentage of early vs total refreshes something that's important? My suggestion for the tooltip would be:

The amount of Atonements that were refreshed earlier than within 3 seconds of the buff expiring. You applied Atonement `200` times in total, `30` (`15`%) of them were refreshes of existing Atonements, and `10` (`5`%) of them were considered early.

This explains what it means, and provides all the info in an easy to understand format.

Comment (nutspanther) [inline src/Parser/DisciplinePriest/CombatLogParser.js]: That makes sense, I was actually having trouble structuring the tooltip to be something that is important. The suggested tooltip makes more sense I will update.

**Comment by MartijnHols** [comment]
Merged, thanks a lot!

**Pull request merged**